linode/manager
# change: [M3-8439] - Tag form events in Linode Create v2

## Description 📝
This PR tags the UI elements from Linode Create v2 that were not shared with the v1 flow (which was tagged in #10722).

## Changes  🔄
See #10722 for an overview how form events are handled.

## Target release date 🗓️
9/3

## Preview 📷
<details>

<summary>
Screenshots of Main Create Flow Events
</summary>


| Component | Analytics Event |
| --- | --- |
| ![GettingStarted](https://github.com/user-attachments/assets/a722ce92-471d-4b11-bfe2-9cbaa2304e77) | ![gettingStartedEvent](https://github.com/user-attachments/assets/0a815720-9e05-4d08-b7ea-3c3aa60bba89) |
| Current: ![DCPricing](https://github.com/user-attachments/assets/e9f7ab52-8692-4819-87bf-9e9c85f9ec2b) Two step:  ![twoStepRegionSelect](https://github.com/user-attachments/assets/c0a53e83-8592-4373-a718-d0c830391833)  | ![twoStepRegionSelectEvent](https://github.com/user-attachments/assets/aa2c37e8-b6ae-40a3-8fb7-e1ce22c5c9af) |
| ![RegionSelect](https://github.com/user-attachments/assets/0ad1865f-0bcc-422e-a13e-26b194ab6051) | ![selectRegion](https://github.com/user-attachments/assets/0338b8a2-8d38-494a-a51d-c8fc57625795) |
| ![ChoosingPlan](https://github.com/user-attachments/assets/3f9ace8c-ea0c-4505-a530-54e5da967d20) | ![planEvent](https://github.com/user-attachments/assets/f25799d8-cd1c-4c4c-8e6f-84942eaf3cc2) |
| ![PGSelect](https://github.com/user-attachments/assets/5b896e70-7bbe-46fc-8c2b-515e6a65284b) | ![changePG](https://github.com/user-attachments/assets/5fa02306-8994-45fd-912b-a97a8a0341de) ![clearPG](https://github.com/user-attachments/assets/12729358-8d25-4c81-b5d9-40f02a6801df) ![createDrawerPG](https://github.com/user-attachments/assets/48fd302e-b631-4e5e-93ee-91f6599dcab6) |
| ![VPCSelect](https://github.com/user-attachments/assets/bc58a037-5697-4f11-bd5b-3ae484056afe) | ![learnMoreVPC](https://github.com/user-attachments/assets/3e47569f-5834-4900-ae40-2490262dd8e1) ![createDrawerVPC](https://github.com/user-attachments/assets/2f24b8a9-7d6b-4edd-9e39-31f31cf71371) VPC is selected: ![changeVPC](https://github.com/user-attachments/assets/bcb8f7c3-5cfd-42c0-8854-a10d8a11540c) VPC is cleared: ![clearVPC](https://github.com/user-attachments/assets/4857caa2-633e-44d9-8f01-2e2d1e2fdbcd) |
| ![FirewallSelect](https://github.com/user-attachments/assets/96462f0a-f175-4520-adba-0f7232dc865f) | ![learnMoreFirewall](https://github.com/user-attachments/assets/d66fdfc2-f88f-4513-92d7-7f736451540b) ![createDrawerFirewall](https://github.com/user-attachments/assets/07366990-6ac6-4f1b-acf5-43378e44f95b) Firewall is selected: ![changeFirewall](https://github.com/user-attachments/assets/5d02c698-5d03-433a-a58e-a5ed07532352)  Firewall is cleared: ![clearFirewall](https://github.com/user-attachments/assets/bb8b378f-cdfd-4586-b44d-4ca137a88b5d)  |
| ![ActionButtons2](https://github.com/user-attachments/assets/898b5af2-7d45-41ca-b150-9fcbbd7ec7eb) | ![viewCodeSnippets](https://github.com/user-attachments/assets/e3512059-4963-4f3b-8222-4b02021528ab) ![createLinode](https://github.com/user-attachments/assets/b4a0e957-912b-4b65-87f2-51389febfa3a) |
| click 'Create Linode' without filling in required fields | ![formErrorEvent](https://github.com/user-attachments/assets/fb28928e-1926-44ae-992b-5b68f3f786b8) |

</details>

<details>

<summary>
Screenshots of PG Branch Flow Events
</summary>

| Component | Analytics Event |
| --- | --- |
| ![Create PG](https://github.com/user-attachments/assets/46f3db48-753a-40a1-930b-15dffedb8c03) | ![createPG](https://github.com/user-attachments/assets/107de803-2110-4c94-aca5-bc5ddea4fd20) |
</details>

<details>

<summary>
Screenshots of Firewall Branch Flow Events
</summary>

| Component | Analytics Event |
| --- | --- |
| ![Assign services Learn More](https://github.com/user-attachments/assets/54a161c9-cf03-4028-b646-53901e330a9e) | ![learnMoreAssignServicesFirewall](https://github.com/user-attachments/assets/4eb9e88c-4eef-47bb-b833-5d17b451225a) |
| ![Create Firewall](https://github.com/user-attachments/assets/86a1cea1-76dd-40da-acd3-4d5a45f28b61) | ![createFirewall](https://github.com/user-attachments/assets/d5c65ed4-21b1-432d-8cb6-57567e7a1667) |
</details>

<details>

<summary>
Screenshots of VPC Branch Flow Events
</summary>

| Component | Analytics Event |
| --- | --- |
| ![Create VPC Learn More](https://github.com/user-attachments/assets/e620a707-9fd7-47ce-a115-a233b8a59bb8) | ![learnMoreCreateVPC](https://github.com/user-attachments/assets/5cb7dc81-f0bf-4a33-bd69-d644714d6583) |
| ![Subnets Learn More](https://github.com/user-attachments/assets/332d2f24-a4e3-4d98-812e-0becf3f63329) | ![learnMoreSubnetVPC](https://github.com/user-attachments/assets/07f4ce3f-eb9a-43af-b91d-5abb038113cc) |
| ![Create VPC](https://github.com/user-attachments/assets/1defe751-2dfe-4c67-83b3-5836aa70c2bc) | ![createVPC](https://github.com/user-attachments/assets/7a92078c-a9f4-4b06-a423-83f8c75bde0f) |
</details>

## How to test 🧪

### Prerequisites
(How to setup test environment)
- You must have the Adobe Analytics environment set in your `.env`.
- Check out this PR, `yarn dev`, and in the browser console, type `_satellite.setDebug(true)`.

### Verification steps
(How to verify changes)
- Below are the two color-coded keys for what events should be tagged in the flow. Please see the Preview section for the events and confirm they are firing as expected.

| Main Flow | Branching Flows (Drawers) |
| --- | --- |
| ![MainFlowInstructions](https://github.com/user-attachments/assets/8560bacd-2d9a-4861-b32a-851227ef9b8b) ![FireOnce](https://github.com/user-attachments/assets/12a41683-b988-431f-95fc-67be80dedd27) | ![Screenshot 2024-08-02 at 10 15 25 AM](https://github.com/user-attachments/assets/83741eb7-9966-4599-b60c-4a2a2823508d) |

## As an Author I have considered 🤔

*Check all that apply*
